web/src/ApplicationEditPage.js

```diff
@@ -950,7 +940,7 @@ class ApplicationEditPage extends React.Component {
 
     const signInUrl = `/login/oauth/authorize?client_id=${this.state.application.clientId}&response_type=code&redirect_uri=${redirectUri}&scope=read&state=casdoor`;
     const maskStyle = {position: "absolute", top: "0px", left: "0px", zIndex: 10, height: "97%", width: "100%", background: "rgba(0,0,0,0.4)"};
-    if (!this.state.application.enablePassword) {
+    if (!this.state.enablePassword) {
```

Comment:
Why define it as a state? Why not use `Setting.isPasswordEnabled(this.state.application)` everywhere directly?

Try to avoid defining new states if possible